src/Image.js

```diff
@@ -1,68 +1,72 @@
 import classNames from 'classnames';
-import React from 'react';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
 
-import { createBootstrapComponent } from './ThemeProvider';
+import { useBootstrapPrefix } from './ThemeProvider';
 
-class Image extends React.Component {
-  static propTypes = {
-    /**
-     * @default 'img'
-     */
-    bsPrefix: PropTypes.string,
-    /**
-     * Sets image as fluid image.
-     */
-    fluid: PropTypes.bool,
+const propTypes = {
+  /**
+   * @default 'img'
+   */
+  bsPrefix: PropTypes.string,
+  /**
+   * Sets image as fluid image.
+   */
+  fluid: PropTypes.bool,
 
-    /**
-     * Sets image shape as rounded.
-     */
-    rounded: PropTypes.bool,
+  /**
+   * Sets image shape as rounded.
+   */
+  rounded: PropTypes.bool,
 
-    /**
-     * Sets image shape as circle.
-     */
-    roundedCircle: PropTypes.bool,
+  /**
+   * Sets image shape as circle.
+   */
+  roundedCircle: PropTypes.bool,
 
-    /**
-     * Sets image shape as thumbnail.
-     */
-    thumbnail: PropTypes.bool,
-  };
+  /**
+   * Sets image shape as thumbnail.
+   */
+  thumbnail: PropTypes.bool,
+};
 
-  static defaultProps = {
-    fluid: false,
-    rounded: false,
-    roundedCircle: false,
-    thumbnail: false,
-  };
+const defaultProps = {
+  fluid: false,
+  rounded: false,
+  roundedCircle: false,
+  thumbnail: false,
+};
 
-  render() {
-    const {
-      bsPrefix,
-      className,
-      fluid,
-      rounded,
-      roundedCircle,
-      thumbnail,
-      ...props
-    } = this.props;
+const Image = React.forwardRef(
+  (
+    { bsPrefix, className, fluid, rounded, roundedCircle, thumbnail, ...props },
+    ref,
+  ) => {
+    bsPrefix = useBootstrapPrefix(bsPrefix, 'img');
 
-    const classes = classNames(
-      fluid && `${bsPrefix}-fluid`,
-      rounded && `rounded`,
-      roundedCircle && `rounded-circle`,
-      thumbnail && `${bsPrefix}-thumbnail`,
+    const classes = useMemo(
```

Comment:
no need to memoize this i think

Reply:
this is a quick calculation, but also it returns a string, so `===` equality checks on the return value will still pass even without memoization

Reply:
Totally right, I pre-optimized here 😬